connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -0,0 +1,364 @@
+            targetAdminClient.createTopics(newTopics, new CreateTopicsOptions()).values().forEach((k, v) -> v.whenComplete((x, e) -> {
+                if (e != null) {
+                    log.warn("Could not create topic {}.", k, e);
+                } else {
+                    log.info("Created remote topic {} with {} partitions.", k, partitionCounts.get(k));
+                }
+            }));
+            targetAdminClient.createPartitions(newPartitions).values().forEach((k, v) -> v.whenComplete((x, e) -> {
+                if (e instanceof InvalidPartitionsException) {
+                    // swallow, this is normal
+                } else if (e != null) {
+                    log.warn("Could not create topic-partitions for {}.", k, e);
+                } else {
+                    log.info("Increased size of {} to {} partitions.", k, partitionCounts.get(k));
+                }
+            }));
+        }
+    }
+
+    private Set<String> listTopics(AdminClient adminClient)
+            throws InterruptedException, ExecutionException {
+        synchronized (this) {
+            return adminClient.listTopics().names().get();
+        }
+    }
+
+    private Collection<AclBinding> listTopicAclBindings()
+            throws InterruptedException, ExecutionException {
+        synchronized (this) {
+            return sourceAdminClient.describeAcls(ANY_TOPIC_ACL).values().get();
+        }
+    }
+
+    private Collection<TopicDescription> describeTopics(Collection<String> topics)
+            throws InterruptedException, ExecutionException {
+        synchronized (this) {
+            return sourceAdminClient.describeTopics(topics).all().get().values();
+        }
+    }
+
+    @SuppressWarnings("deprecation")
+    // use deprecated alterConfigs API for broker compatibility back to 0.11.0
+    private void updateTopicConfigs(Map<String, Config> topicConfigs)
+            throws InterruptedException, ExecutionException {
+        Map<ConfigResource, Config> configs = topicConfigs.entrySet().stream()
+            .filter(x -> shouldReplicateTopicConfigurationProperty(x.getKey()))
+            .collect(Collectors.toMap(x ->
+                new ConfigResource(ConfigResource.Type.TOPIC, x.getKey()), x -> x.getValue()));
+        log.trace("Syncing configs for {} topics.", configs.size());
+        synchronized (this) {
+            targetAdminClient.alterConfigs(configs).values().forEach((k, v) -> v.whenComplete((x, e) -> {
+                if (e != null) {
+                    log.warn("Could not alter configuration of topic {}.", k.name(), e);
+                }
+            }));
+        }
+    }
+
+    private void updateTopicAcls(List<AclBinding> bindings)
+            throws InterruptedException, ExecutionException {
+        log.trace("Syncing {} topic ACL bindings.", bindings.size());
+        synchronized (this) {
+            targetAdminClient.createAcls(bindings).values().forEach((k, v) -> v.whenComplete((x, e) -> {
```

Comment:
As per the KIP the mirrored topic should be Writable by Mirror Maker only, how is this ensured? I couldn't find a filter rejecting any WRITE acls to the topic in this call?